src/App.tsx

```diff
@@ -1,24 +1,200 @@
+      .then((data) => {
+        setTodos(data);
+      })
+      .catch(() => {
+        showErrorNotification('Unable to load todos');
+      });
+  }, []);
+
+  const addNewTodo = (newTodo: Todo) => {
+    setTodos([...todos, newTodo]);
+  };
+
+  const handleUpdate = (updatedtodo: Todo) => {
+    setProcesingTodoIds(prev => [...prev, updatedtodo.id]);
+
+    updateTodo(updatedtodo)
+      .then((todoFromserver) => {
+        setTodos((corentTodos) => {
+          const newTodos = [...corentTodos];
+          const index = newTodos.findIndex(
+            (todo) => todo.id === updatedtodo.id,
+          );
+
+          newTodos.splice(index, 1, todoFromserver);
+
+          return newTodos;
+        });
+      })
+      .catch(() => {
+        showErrorNotification('Unable to update a todo');
+      })
+      .finally(() => {
+        setProcesingTodoIds(prev => prev
+          .filter(id => id !== updatedtodo.id));
+      });
+  };
+
+  const togleCheck = (todo: Todo) => {
+    const chekedtodo = { ...todo, completed: !todo.completed };
+
+    handleUpdate(chekedtodo);
+  };
+
+  const isAllCompleted = todos.every((todo) => todo.completed); //
+
+  const toggleAll = () => {
+    const todosToUpdate = todos.filter((todo) => (isAllCompleted
+      ? todo.completed
+      : !todo.completed));
+
+    const updatePromises = todosToUpdate.map((todo) => (
+      handleUpdate({
+        ...todo,
+        completed: !isAllCompleted,
+      })));
+
+    return Promise.all(updatePromises);
+  };
+
+  //
+
+  const toDelete = (id: number) => {
+    const afterDeleteTodo = todos.filter((todo) => todo.id !== id);
+
+    setTodos(afterDeleteTodo);
+  };
+
+  const handleDeletedTodo = (id: number) => {
+    setProcesingTodoIds((prev) => [...prev, id]);
+
+    deleteTodo(id)
+      .then(() => {
+        toDelete(id);
+      })
+      .catch(() => {
+        showErrorNotification('Unable to delete a todo');
+      })
+      .finally(() => {
+        setProcesingTodoIds((prev) => prev
+          .filter((processingId) => processingId !== id));
+      });
+  };
+  //
+
+  const filtredTodo = todos.filter((todo) => {
+    switch (filtredByStatus) {
+      case Status.active:
+        return !todo.completed;
+      case Status.completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {todos.length > 0 && (
```

Comment:
```suggestion
          {!!todos.length && (
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**